src/main/java/org/opentripplanner/openstreetmap/wayproperty/specifier/Condition.java

```diff
@@ -160,18 +203,34 @@ public boolean isExtendedKeyMatch(OSMWithTags way, String exKey) {
       var maybeInt = way.getTagAsInt(exKey, ignored -> {});
       return maybeInt.isPresent() && maybeInt.getAsInt() >= lower && maybeInt.getAsInt() <= upper;
     }
+
+    @Override
+    public String toString() {
+      return "%s > %s < %s".formatted(lower, key, upper);
+    }
   }
 
-  record EqualsAnyIn(String key, String... values) implements Condition {
+  /**
+   * Selects a tag which has one of a set of given values.
+   */
+  record OneOf(String key, String... values) implements Condition {
     @Override
     public boolean isExtendedKeyMatch(OSMWithTags way, String exKey) {
       return Arrays.stream(values).anyMatch(value -> way.isTag(exKey, value));
     }
+
+    @Override
+    public String toString() {
+      return "%s one of [%s]".formatted(key, String.join(", ", values));
+    }
   }
 
-  record EqualsAnyInOrAbsent(String key, String... values) implements Condition {
+  /**
+   * Selects a tag which has one of a set of given values or the tag is absent.
```

Comment:
This is slightly difficult to read but I couldn't come up with a better way to formulate what this does.

Reply:
I've tried to improve it here: https://github.com/opentripplanner/OpenTripPlanner/pull/5929/commits/a9441a7060ba7e9c56ad667280f83933288323ea